docker/dev-entrypoint.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -x
+set -e
+
+export USER=$(whoami) # $USER isn't set!?
+
+APP_DEV_DIR="/home/$USER/ondemand/dev"
+OOD_DEV_DIR="/var/www/ood/apps/dev/$USER"
+
+sudo su root <<MKDEV
+  mkdir -p $OOD_DEV_DIR 
+  chmod 755 $OOD_DEV_DIR
+  cd $OOD_DEV_DIR
+  ln -s $APP_DEV_DIR gateway
+MKDEV
+
+/opt/ood/ood-portal-generator/sbin/update_ood_portal
+
+if [ -n "$OOD_STATIC_USER" ] && [ -f "$OOD_STATIC_USER" ]; then
```

Comment:
Where is `OOD_STATIC_USER` getting set or how does that get set?  Why not just do this?

```
sed -i "s/ood/$USER/g" /etc/ood/dex/config.yaml
```

Reply:
It's a mounted file and environment variable pointing to it.  It's to change the password too. There's an attack vector here where if you're you (you've kept your uid) and you've mounted your home directory (or some parts of it) and started a webserver with a known password - well you've just played yourself.  So I put this in here to get a new password.

So anyway I have these mount points 
```
CFG="$CFG -v $OOD_CFG/static_user.yml:/etc/ood/dex/static_user.yml"
CFG="$CFG -e OOD_STATIC_USER=/etc/ood/dex/static_user.yml"
```

and a little helper to write.
```shell
#!/bin/bash

export OOD_CTR_PASSWORD=$(bcrypt $1)

cat <<EOF > static_user.yml
enablePasswordDB: true
staticPasswords:
- email: "$USER@localhost"
  hash: "$OOD_CTR_PASSWORD"
  username: "$USER"
  userID: "71e63e31-7af3-41d7-add2-575568f4525f"
EOF

unset SOME_PASSWORD
```